app/src/main/java/com/android/unio/model/event/EventRepository.kt

```diff
@@ -0,0 +1,26 @@
+package com.android.unio.model.event
+
+import com.google.firebase.Timestamp
+
+interface EventRepository {
+  fun getEventsOfAssociation(
+      association: String,
+      onSuccess: (List<Event>) -> Unit,
+      onFailure: (Exception) -> Unit
+  )
+
+  fun getNextEventsFromDateToDate(
+      startDate: Timestamp,
+      endDate: Timestamp,
+      onSuccess: (List<Event>) -> Unit,
+      onFailure: (Exception) -> Unit
+  )
+
+  fun getEvents(onSuccess: (List<Event>) -> Unit, onFailure: (Exception) -> Unit)
+
+  fun getNewUid(): String
+
+  fun addEvent(event: Event, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
+
```

Comment:
Need to ask to the coaches about the updateEvent function: is it really necessary to write one in the repository since it is essentially a copy of the addEvent function of the repository or not.